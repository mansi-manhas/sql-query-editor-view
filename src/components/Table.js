import React from "react";
import { useTable, useBlockLayout } from "react-table";
import { FixedSizeList } from "react-window";

import { TableWrapper } from "./styled.style";
import { getScrollbarWidth } from "./utils";

const Table = ({ columns, data }) => {

  const defaultColumn = React.useMemo(
    () => ({
      width: 200,
    }),
    []
  );

  const {
    getTableProps,
    getTableBodyProps,
    headerGroups,
    rows,
    totalColumnsWidth,
    prepareRow,
  } = useTable(
    {
      columns: columns ?? [],
      data: data ?? [],
      defaultColumn,
    },
    useBlockLayout,
  );

  const scrollBarSize = React.useMemo(() => getScrollbarWidth(), []);

  const RenderRow = React.useCallback(
    ({ index, style }) => {
      const row = rows[index];
      prepareRow(row);
      return (
        <div
          {...row.getRowProps({
            style,
          })}
          className="tr"
        >
          {row.cells.map((cell) => {
            return (
              <div {...cell.getCellProps()} className="td">
                {cell.render("Cell")}
              </div>
            );
          })}
        </div>
      );
    },
    [prepareRow, rows]
  );

  return (
    <TableWrapper>
      <div {...getTableProps()} className="table">
        <div>
          {headerGroups.map((headerGroup) => (
            <div {...headerGroup.getHeaderGroupProps()} className="tr">
              {headerGroup.headers.map((column) => (
                <div {...column.getHeaderProps()} className="th">
                  {column.render("Header")}
                </div>
              ))}
            </div>
          ))}
        </div>
        <div {...getTableBodyProps()}>
          <FixedSizeList
            height={400}
            itemCount={rows.length}
            itemSize={50}
            width={totalColumnsWidth + scrollBarSize}
          >
            {RenderRow}
          </FixedSizeList>
        </div>
      </div>
    </TableWrapper>
  );
};

export default Table;
