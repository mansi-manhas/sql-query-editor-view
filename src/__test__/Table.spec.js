import React from "react";
import renderer from "react-test-renderer";
import Table from "../components/Table";

describe("Jest Snapshot testing suite", () => {
  it("Matches DOM Snapshot", () => {
    const domTree = renderer.create(<Table />).toJSON();
    expect(domTree).toMatchSnapshot();
  });
});
