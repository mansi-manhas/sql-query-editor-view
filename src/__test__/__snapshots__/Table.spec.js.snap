// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Jest Snapshot testing suite Matches DOM Snapshot 1`] = `
<div
  className="sc-bcXHqe duipyf"
>
  <div
    className="table"
    role="table"
  >
    <div />
    <div
      role="rowgroup"
    >
      <div
        onScroll={[Function]}
        style={
          Object {
            "WebkitOverflowScrolling": "touch",
            "direction": "ltr",
            "height": 400,
            "overflow": "auto",
            "position": "relative",
            "width": 0,
            "willChange": "transform",
          }
        }
      >
        <div
          style={
            Object {
              "height": 0,
              "pointerEvents": undefined,
              "width": "100%",
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;
